sdk/src/tx/whileValidTxSender.ts

```diff
@@ -35,6 +35,19 @@ export class WhileValidTxSender extends BaseTxSender {
 	>();
 	blockhashCommitment: Commitment;
 
+	useBlockHeightOffset = true;
+
+	private async checkAndSetUseBlockHeightOffset() {
+		this.connection.getVersion().then((version) => {
+			const solanaCoreVersion = version['solana-core'];
+			if (parseInt(solanaCoreVersion.split('.')[0]) >= 2) {
```

Comment:
maybe undefined guard here after `solanaCoreVersion` just incase it comes back where that member is undefined